spec/controllers/communicarts_controller_spec.rb

```diff
@@ -303,7 +303,17 @@
         expect { put 'approval_response', @json_approval_params_with_token }.to raise_error(AuthenticationError)
       end
 
-      skip 'marks a token as used'
+      it 'marks a token as used' do
+        expect(ApiToken).to receive(:find_by).with(access_token: "5a4b3c2d1ee1d2c3b4a5").and_return(token)
+        expect_any_instance_of(Approval).to receive(:update_attributes)
+        approver
+
+        put 'approval_response', @json_approval_params_with_token
+
+        expect(response).to redirect_to(cart_path(cart))
+        token.reload
+        expect(token.used_at).to_not eq(nil)
```

Comment:
Might be worth extending this just a bit further to show that if a user tries to log in with this again, they cannot. I think that's implied by the token, but I'm not sure.

Related question - if a user finds the same email again, and they click one of the links, can they still get to the cart (after logging in)?